src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt

```diff
@@ -236,69 +367,56 @@ open class HiddenSecretsPlugin : Plugin<Project> {
-                }
-                //Resolve package name for C++ from the one used in Kotlin file
-                var kotlinPackage = Utils.getKotlinFilePackage(secretsKotlin)
-                if (kotlinPackage.isEmpty()) {
-                    println("Empty package in $KOTLIN_FILE_NAME")
-                    kotlinPackage = packageName
-                }
+                hideSecret(
+                    keyName = keyName,
+                    packageName = packageName,
+                    obfuscatedKey = obfuscatedKey
+                )
+            }
+        }
 
-                //Add obfuscated key in C++ code
-                val secretsCpp = getCppDestination("secrets.cpp")
-                if (secretsCpp.exists()) {
-                    var text = secretsCpp.readText(Charset.defaultCharset())
-                    if (text.contains(obfuscatedKey)) {
-                        println("⚠️ Key already added in C++ !")
-                    }
-                    if (text.contains(KEY_PLACEHOLDER)) {
-                        //Edit placeholder key
-                        //Replace package name
-                        text = text.replace(PACKAGE_PLACEHOLDER, Utils.getSnakeCasePackageName(kotlinPackage))
-                        //Replace key name
-                        text = text.replace("YOUR_KEY_NAME_GOES_HERE", keyName)
-                        //Replace demo key
-                        text = text.replace(KEY_PLACEHOLDER, obfuscatedKey)
-                        secretsCpp.writeText(text)
-                    } else {
-                        //Add new key
-                        text += CodeGenerator.getCppCode(kotlinPackage, keyName, obfuscatedKey)
-                        secretsCpp.writeText(text)
-                    }
-                } else {
-                    error("Missing C++ file, please run gradle task : $TASK_COPY_CPP")
+        /**
+         * Clean all secret hidden keys in your project and obfuscate all keys from the properties file.
+         */
+        project.task(TASK_HIDE_SECRET_FROM_PROPERTIES_FILE)
+        {
+            this.group = TASK_GROUP
+            this.description = "Re-generate and obfuscate keys from properties file and add it to your Android project"
+            dependsOn(TASK_UNZIP_HIDDEN_SECRETS)
+
+            doLast {
+                //Create a clean copy of dependency files
+                copyCppFiles(true)
+                copyKotlinFile(true)
+
+                val packageName = getPackageNameParam()
+                val propsFile = getPropertiesFile()
+                val props = getPropertiesFromFile(propsFile = propsFile)
+                println("Generating secrets from props: ${propsFile.path}")
+                props.entries.forEach { entry ->
+                    val keyName = entry.key as String
+                    val obfuscatedKey = Utils.encodeSecret(entry.value as String, packageName)
+                    hideSecret(
```

Comment:
Not need to repeat names and vars: 
`hideSecret(keyName, packageName, obfuscatedKey)`